### PR TITLE
Remove float64

### PR DESCRIPTION
**Pull request by rezaarezvan**
Removes all `float64` instances along with all associated tests with them.

### CODE REVIEW

File: test/external/external_test_onnx_backend.py [hunk 1]
```diff
@@ -60,7 +60,6 @@ def supports_device(cls, device: str) -> bool:
 backend_test.exclude('uint64')
 backend_test.exclude('int8')
 backend_test.exclude('int16')
-backend_test.exclude('float64')
```
Comment (geohot) [inline test/external/external_test_onnx_backend.py:63]: Doesn't this exclude float64?

**Comment by tinyb0t** [comment]
Changes made in `tinygrad/`:
```
------------------------------------------------------------
files                             insertions       deletions
------------------------------------------------------------
tinygrad/codegen/assembly_rdna.py          1               1
tinygrad/codegen/cstyle.py                 0               1
tinygrad/codegen/llvmir.py                 1               5
tinygrad/helpers.py                        1               2
tinygrad/runtime/lib.py                    1               1
tinygrad/runtime/ops_gpu.py                1               2
tinygrad/runtime/ops_metal.py              1               2
tinygrad/runtime/ops_torch.py              1               1
------------------------------------------------------------
total                                      7              15
------------------------------------------------------------
lines added in the tinygrad folder: -8
```

File: test/test_tensor.py [hunk 2]
```diff
@@ -105,41 +105,6 @@ def test_dropout(self):
     expected = n * (1 - rate)
     np.testing.assert_allclose(non_zeros, expected, rtol=2e-3)
 
-  #@unittest.skipUnless(Device.DEFAULT == Device.CPU, "float64 not supported on GPU")
-  @unittest.skip("float64 support broken")
```
Comment (posix4e) [inline test/test_tensor.py:109]: If we remove this shouldn't we remove extra/gradcheck.py

File: test/external/external_test_onnx_backend.py (hunk 1 above)
Comment (rezaarezvan) [reply test/external/external_test_onnx_backend.py:63]: Oh yeah - my bad, I'll revert that

File: test/test_tensor.py (hunk 2 above)
Comment (rezaarezvan) [reply test/test_tensor.py:109]: > If we remove this shouldn't we remove extra/gradcheck.py

Why?

Comment (geohot) [reply test/test_tensor.py:109]: Can you fix this test instead of removing it?

Comment (rezaarezvan) [reply test/test_tensor.py:109]: > Can you fix this test instead of removing it?

Sure, I'll see what I can do

Comment (geohot) [reply test/test_tensor.py:109]: would also merge with this test skipped and not removed

Comment (rezaarezvan) [reply test/test_tensor.py:109]: > would also merge with this test skipped and not removed

I think it's cleaner if we merge now. I'll open a new PR that fixes those tests instead

**Comment by geohot** [comment]
Thanks!

**Comment by Deferf** [comment]
Now when trying to import a random array from Numpy it returns `KeyError: 'float64'`. 

```
from tinygrad.tensor import Tensor
import numpy as np

X = np.random.random((3,3))
Tensor(X)
```

Are we expected to change the dtype? Or is it better to add a casting operation to fp32?

**Comment by geohotstan** [comment]
Also for `numpy_fxn_for_op[BinaryOps.MAX]` in `ops_cpu.py`: 
```
In [33]: np.promote_types(np.float32, np.int32)
Out[33]: dtype('float64')

To reproduce:
In [1]: from tinygrad.lazy import Device                                                                                                                                                                                                          
In [2]: from tinygrad.tensor import Tensor                                                                                                                                                        
In [3]: from tinygrad.helpers import dtypes                                                                                                                                                                                                       
In [4]: Device.DEFAULT = "CPU"                                                                                                                   
In [5]: (Tensor.arange(3,dtype=dtypes.int32) == Tensor([1.0,1.0,2.412], dtype=dtypes.float32)).realize()
KeyError: 'float64'
```
torch on the other hand handles promote_types as promoting to float32 in these cases.

**Pull request merged**